rust/blockstore/src/arrow/block/value/int32array_value.rs

```diff
@@ -43,6 +46,34 @@ impl ArrowWriteableValue for Vec<u32> {
     fn get_delta_builder() -> BlockStorage {
         BlockStorage::VecUInt32(SingleColumnStorage::new())
     }
+
+    fn get_value_builder() -> Self::ValueBuilder {
+        ListBuilder::new(UInt32Builder::new())
```

Comment:
Capacity is no longer pre-allocated as the type of size hints that would have to be provided across value types is not homogeneous, making implementation a bit tricky. I think this is ok since we always `.slice()` the columns during `.finish()`, but can attempt to add it back if we need it.

I'm not yet sure what performance impact this change has; a future PR in this stack will add benchmarks for blockfile writers and if pre-allocation makes a significant performance difference I can attempt to add it then.

Reply:
Having the concept of a size tracker in general would be helpful as it would also allow us to unify `DataRecordStorage` and `SingleColumnStorage`.

Reply:
decision: can add a GAT for a size hint type